packages/module/src/utils.ts

```diff
@@ -36,3 +38,80 @@ export const parseMaybeNumericString = (numOrStr: string | undefined | null): st
   const num = Number(numOrStr);
   return isNaN(num) ? numOrStr : num;
 };
+
+export const mergeArgs = <
```

Comment:
nit: This would ideally be unit tested right now when introduced.

Reply:
Good call. Will do.

Reply:
Added unit tests